Allow the flipped state to be passed via props

muyiwa-felixson opened the pull request.

Commit:
Allow the flipped state to be passed via props

Commit:
Update FlipCard.js

Commit:
Removed TouchableOpacity from Component Container

TouchOpacity can be applied to a parent container, leave the decision to the developer/User. 
Added option to allow the flip component receive the onpress property, or pass it via props

calcazar commented on FlipCard.js:
```diff
@@ -50,12 +54,12 @@ class FlipCard extends Component{
         });
         return (
-            <TouchableOpacity onPress={this._flipToggleCard} style={styles.animatedContainer}>
+            <View onPress={this._flipToggleCard} style={styles.animatedContainer}>
```
This should remain as TouchableOpacity. React-Native doesn't officially support onPress for Views. https://facebook.github.io/react-native/docs/view.html

muyiwa-felixson replied:
That works too, I would add the option to pass the opacity props to the component.

calcazar reviewed (changes requested):
I would just chance the view back to a touchable opacity. If you wanted no opacity I would probably create a prop that controls the opacity level. Maybe even a boolean.

Commit:
Reverted the parent view back to touchable opacity

Parent view reverted back to touch opacity,
Added optional touch opacity property to the parent view.

muyiwa-felixson commented:
@calcazar kindly review the changes

calcazar commented on FlipCard.js:
```diff
@@ -47,19 +47,24 @@ class FlipCard extends Component{
-
+    _getOpacity() {
+        if (!this.props.activeOpacity) {
+            return 1;
```
I would suggest we return the default of 0.2 and the let the user override the default with 1. Everything else looks good~

calcazar reviewed (changes requested):
I would make the default to 0.2 and then allow users to override that with the prop

Commit:
Defaults touch opacity to 0.2

My reason for making the initial 1 earlier was to take care of the case where the flip trigger is not the component itself. for example
<TouchableOpacity onPress={() => {  this.setState({ flipDetail: true }); }} >
    <Text>View Card Detail</Text>
</TouchableOpacity>

Commit:
Quick correction on line 54

Commit:
Passed appropriate value to activeOpacity prop

Commit:
Reverted last line back to ES5

muyiwa-felixson commented:
@calcazar Note I added prop type validation
